compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt

```diff
@@ -477,14 +477,33 @@ class ComposeScene internal constructor(
     /**
      * Returns the current content size
      */
+    @Deprecated("Will be removed in 1.7", replaceWith = ReplaceWith("calculateContentSize()"))
     val contentSize: IntSize
         get() {
-            check(!isClosed) { "ComposeScene is closed" }
             val mainOwner = mainOwner ?: return IntSize.Zero
-            mainOwner.measureAndLayout()
-            return mainOwner.contentSize
+            return contentSizeInConstraints(mainOwner.constraints)
         }
 
+    /**
+     * Returns the current content size in infinity constraints.
+     *
+     * @throws IllegalStateException when [ComposeScene] content has lazy layouts without maximum size bounds
+     * (e.g. LazyColumn without maximum height).
+     */
+    fun calculateContentSize(): IntSize {
+        return contentSizeInConstraints(Constraints())
+    }
+
+    /**
+     * Provides [ComposeScene] content size in given [constraints]
+     *
+     * This function doesn't change current [ComposeScene] scene constraints.
+     */
+    private fun contentSizeInConstraints(constraints: Constraints): IntSize {
```

Comment:
Let's inline the content into `calculateContentSize`. A separate function doesn't add any value now